airbyte-webapp/src/components/base/Input/Input.tsx

```diff
@@ -23,67 +24,88 @@ export type InputProps = {
+  position: relative;
   background: ${(props) => getBackgroundColor(props)};
-  color: ${({ theme }) => theme.textColor};
-  caret-color: ${({ theme }) => theme.primaryColor};
-
-  &::placeholder {
-    color: ${({ theme }) => theme.greyColor40};
-  }
+  border: 1px solid ${(props) => (props.error ? props.theme.dangerColor : props.theme.greyColor0)};
+  border-radius: 4px;
 
   &:hover {
     background: ${({ theme, light }) => (light ? theme.whiteColor : theme.greyColor20)};
     border-color: ${(props) => (props.error ? props.theme.dangerColor : props.theme.greyColor20)};
   }
 
-  &:focus {
+  &.input-container--focused {
     background: ${({ theme, light }) => (light ? theme.whiteColor : theme.primaryColor12)};
     border-color: ${({ theme }) => theme.primaryColor};
   }
+`;
+
+const InputComponent = styled.input<InputProps & { isPassword?: boolean }>`
+  outline: none;
+  width: ${({ isPassword }) => (isPassword ? "calc(100% - 22px)" : "100%")};
+  padding: 7px 8px 7px 8px;
+  font-size: 14px;
+  line-height: 20px;
+  font-weight: normal;
+  border: none;
+  background: none;
+  color: ${({ theme }) => theme.textColor};
+  caret-color: ${({ theme }) => theme.primaryColor};
+
+  &::placeholder {
+    color: ${({ theme }) => theme.greyColor40};
+  }
 
   &:disabled {
     pointer-events: none;
     color: ${({ theme }) => theme.greyColor55};
   }
 `;
 
-const Container = styled.div`
-  width: 100%;
-  position: relative;
-`;
-
 const VisibilityButton = styled(Button)`
   position: absolute;
-  right: 2px;
-  top: 7px;
+  right: 0px;
+  top: 0;
+  display: flex;
+  height: 100%;
+  width: 30px;
+  align-items: center;
+  justify-content: center;
+  border: none;
 `;
 
 const Input: React.FC<InputProps> = (props) => {
   const [isContentVisible, setIsContentVisible] = useState(false);
-
-  if (props.type === "password") {
-    return (
-      <Container>
-        <InputComponent {...props} type={isContentVisible ? "text" : "password"} />
-        {props.disabled ? null : (
-          <VisibilityButton iconOnly onClick={() => setIsContentVisible(!isContentVisible)} type="button">
-            <FontAwesomeIcon icon={isContentVisible ? faEyeSlash : faEye} />
-          </VisibilityButton>
-        )}
-      </Container>
-    );
-  }
-
-  return <InputComponent {...props} />;
+  const [focused, toggleFocused] = useToggle(false);
+
+  const isPassword = props.type === "password";
+  const isVisibilityButtonVisible = isPassword && !props.disabled;
+  const onInputFocusChange = () => toggleFocused();
+
+  return (
+    <InputContainer {...props} className={focused ? "input-container--focused" : undefined}>
+      <InputComponent
+        {...props}
+        type={!isPassword || isContentVisible ? "text" : "password"}
+        isPassword={isPassword}
+        onFocus={onInputFocusChange}
+        onBlur={onInputFocusChange}
+        data-testid="input"
+      />
+      {isVisibilityButtonVisible ? (
+        <VisibilityButton
```

Comment:
Since this is an icon only button, could we still attach an `aria-label` to it with "Toggle password visibility" (or the like)